### PR TITLE
Fix tests: issue #268

### PR DESCRIPTION
**Pull request by Ingvord**

### CODE REVIEW

**Comment by taurel** [comment]
Hi Igor,

Could you send me please the result of the command "DbGetDataForServerCache" on your db server?
The input args should be "dscache/test" and the host name where the devTest DS is running

Cheers

Emmanuel

**Comment by Ingvord** [comment]
@taurel here is the output requested:

```
Command: sys/database/2/DbGetDataForServerCache
Duration: 74 msec
Argin: dscache/test, 172.17.0.3
Output argument(s) :
array length: 67
[0]	 dserver/dscache/test
[1]	 nada
[2]	 0
[3]	 DsCache/test
[4]	 nada
[5]	 0
[6]	 0
[7]	 DServer
[8]	 notifd/factory/172.17.0.3
[9]	 Not Found
[10]	 dserver/dscache/test
[11]	 Not Found
[12]	 DServer
[13]	 1
[14]	 AllowedAccessCmd
[15]	 10
[16]	 QueryClass
[17]	 QueryDevice
[18]	 EventSubscriptionChange
[19]	 DevPollStatus
[20]	 GetLoggingLevel
[21]	 GetLoggingTarget
[22]	 QueryWizardDevProperty
[23]	 QueryWizardClassProperty
[24]	 QuerySubDevice
[25]	 ZMQEventSubscriptionChange
[26]	 Default
[27]	 0
[28]	 dserver/dscache/test
[29]	 0
[30]	 dscache/test
[31]	 2
[32]	 CacheTest1
[33]	 0
[34]	 CacheTest1
[35]	 0
[36]	 CacheTest1
[37]	 2
[38]	 test/cache1/1
[39]	 test/cache1/2
[40]	 test/cache1/1
[41]	 1
[42]	 tst_property
[43]	 1
[44]	 15
[45]	 test/cache1/1
[46]	 0
[47]	 test/cache1/2
[48]	 1
[49]	 tst_property
[50]	 1
[51]	 25
[52]	 test/cache1/2
[53]	 0
[54]	 CacheTest2
[55]	 0
[56]	 CacheTest2
[57]	 0
[58]	 CacheTest2
[59]	 1
[60]	 test/cache2/1
[61]	 test/cache2/1
[62]	 0
[63]	 test/cache2/1
[64]	 0
[65]	 CtrlSystem
[66]	 0
```

**Comment by Ingvord** [comment]
@taurel here is the size test output:

```
28: Test command: /home/tango/src/build/cpp_test_suite/old_tests/size

28: Test timeout computed to be: 1500

28:    Size of Attr should be 76 and is 128 --> NOK !!!!!!!!!!!

28:    Size of SpectrumAttr should be 84 and is 144 --> NOK !!!!!!!!!!!

28:    Size of ImageAttr should be 92 and is 160 --> NOK !!!!!!!!!!!

28:    Size of DeviceImpl should be 1080 and is 1760 --> NOK !!!!!!!!!!!

28:    Size of Device_2Impl should be 1088 and is 1776 --> NOK !!!!!!!!!!!

28:    Size of DeviceClass should be 236 and is 416 --> NOK !!!!!!!!!!!

28:    Size of Util should be 908 and is 1616 --> NOK !!!!!!!!!!!

28:    Size of Attribute should be 984 and is 2536 --> NOK !!!!!!!!!!!

28:    Size of WAttribute should be 1420 and is 3200 --> NOK !!!!!!!!!!!

28:    Size of MultiAttribute should be 40 and is 80 --> NOK !!!!!!!!!!!

28:    Size of Command should be 40 and is 72 --> NOK !!!!!!!!!!!

28:    Size of TemplCommand should be 60 and is 112 --> NOK !!!!!!!!!!!

28:    Size of TemplCommandIn should be 72 and is 136 --> NOK !!!!!!!!!!!

28:    Size of TemplCommandOut should be 72 and is 136 --> NOK !!!!!!!!!!!

28:    Size of TemplCommandInOut should be 72 and is 136 --> NOK !!!!!!!!!!!

28:    Size of DeviceProxy should be 360 and is 576 --> NOK !!!!!!!!!!!

28:    Size of DeviceData should be 16 and is 32 --> NOK !!!!!!!!!!!

28:    Size of DeviceDataHistory should be 48 and is 80 --> NOK !!!!!!!!!!!

28:    Size of DeviceAttribute should be 120 and is 200 --> NOK !!!!!!!!!!!

28:    Size of DeviceAttributeHistory should be 144 and is 232 --> NOK !!!!!!!!!!!

28:    Size of ApiUtil should be 176 and is 312 --> NOK !!!!!!!!!!!

28:    Size of Database should be 380 and is 624 --> NOK !!!!!!!!!!!

28:    Size of DbDevice should be 20 and is 32 --> NOK !!!!!!!!!!!

28:    Size of DbClass should be 20 and is 32 --> NOK !!!!!!!!!!!

28:    Size of DbServer should be 20 and is 32 --> NOK !!!!!!!!!!!

28:    Size of DbDatum should be 32 and is 56 --> NOK !!!!!!!!!!!

28:    Size of DbData should be 12 and is 24 --> NOK !!!!!!!!!!!

28:    Size of CmdDoneEvent should be 24 and is 48 --> NOK !!!!!!!!!!!

28:    Size of AttrReadEvent should be 24 and is 48 --> NOK !!!!!!!!!!!

28:    Size of AttrWrittenEvent should be 20 and is 40 --> NOK !!!!!!!!!!!

28:    Size of CallBack should be 8 and is 16 --> NOK !!!!!!!!!!!

28:    Size of EncodedAttribute should be 32 and is 56 --> NOK !!!!!!!!!!!

28:    Size of Group should be 80 and is 152 --> NOK !!!!!!!!!!!

28:    Size of MultiAttrProp<T> should be 296 and is 592 --> NOK !!!!!!!!!!!

28:    Size of AttrProp<T> should be 16 and is 32 --> NOK !!!!!!!!!!!

28:    Size of DoubleAttrProp<T> should be 24 and is 48 --> NOK !!!!!!!!!!!

28:    Size of DevicePipe should be 68 and is 112 --> NOK !!!!!!!!!!!

28:    Size of DevicePipeBlob should be 48 and is 80 --> NOK !!!!!!!!!!!

28/59 Test #28: old_tests::size ..................   Passed    0.02 sec
```

File: cpp_test_suite/event/stateless_sub.cpp [hunk 1]
```diff
@@ -96,7 +96,7 @@ int main(int argc, char **argv)
 // Wait for connection and event
 //
 
-		Tango_sleep(40);
+		Tango_sleep(50);
```
Comment (Ingvord) [inline cpp_test_suite/event/stateless_sub.cpp]: Strangely enough this test always passes in RELEASE mode but not in DEBUG. I suspect some optimizations take place and therefore the code is fast enough to catch last non-error callback.

**Pull request merged**